doc-content/drools-docs/src/main/asciidoc/DecisionEngine/cep-event-streams-proc.adoc

```diff
@@ -0,0 +1,60 @@
+[id='cep-event-streams-proc_{context}']
+
+= Declaring entry points for rule data
+
+You can declare an entry point (event stream) for events so that the {DECISION_ENGINE} uses data from only that entry point to evaluate the rules. You can declare an entry point either implicitly by referencing it in DRL rules or explicitly in your Java application.
+
+.Procedure
+Use one of the following methods to declare the entry point:
+
+* In the DRL rule file, specify `from entry-point "<name>"` for the inserted fact:
++
+--
+.Authorize withdrawal rule with "ATM Stream" entry point
+[source]
+----
+rule "Authorize withdrawal"
+when
+  WithdrawRequest($ai : accountId, $am : amount) from entry-point "ATM Stream"
+  CheckingAccount(accountId == $ai, balance > $am)
+then
+  // Authorize withdrawal.
+end
+----
+
+.Apply fee rule with "Branch Stream" entry point
+[source]
+----
+rule "Apply fee on withdraws on branches"
+when
+  WithdrawRequest($ai : accountId, processed == true) from entry-point "Branch Stream"
+  CheckingAccount(accountId == $ai)
+then
+  // Apply a $2 fee on the account.
+end
+----
+
+Both example DRL rules from a banking application insert the event `WithdrawalRequest` with the fact `CheckingAccount`, but from different entry points. At compile time, the {DECISION_ENGINE} evaluates the `Authorize withdrawal` rule using data from only the `"ATM Stream"` entry point, and evaluates the `Apply fee` rule using data from only the `"Branch Stream"` entry point. Any events inserted into the `"ATM Stream"` can never match patterns for the `"Apply fee"` rule, and any events inserted into the `"Branch Stream"` can never match patterns for the `"Authorize withdrawal rule"`.
+--
+
+* In the Java application configuration code, use the `EntryPoint` object to specify the entry point and insert facts into that entry point accordingly:
```

Comment:
Please replace this with "In the Java application code, use `getEntryPoint()` method to specify and obtain an `EntryPoint` object and insert facts into that entry point accordingly."

Does it make sense?

Reply:
Yep, done.